File: common/src/main/java/net/impleri/playerskills/events/ClientSkillsUpdatedEvent.java
package net.impleri.playerskills.events;

import dev.architectury.event.Event;
import dev.architectury.event.EventActor;
import dev.architectury.event.EventFactory;
import net.impleri.playerskills.api.Skill;

import java.util.List;

public record ClientSkillsUpdatedEvent(List<Skill<?>> next,
                                       List<Skill<?>> prev) {
    public static final Event<EventActor<ClientSkillsUpdatedEvent>> EVENT = EventFactory.createEventActorLoop();
}

File: common/src/main/kotlin/net/impleri/playerskills/events/ClientSkillsUpdatedEvent.kt
package net.impleri.playerskills.events

import dev.architectury.event.Event
import dev.architectury.event.EventFactory
import net.impleri.playerskills.api.Skill
import java.util.function.Consumer

@JvmRecord
data class ClientSkillsUpdatedEvent(
  val next: List<Skill<*>>,
  val prev: List<Skill<*>>,
  val forced: Boolean,
) {
  companion object {
    val EVENT: Event<Consumer<ClientSkillsUpdatedEvent>> = EventFactory.createConsumerLoop()
  }
}
